app/models/revise_entries.rb

```diff
@@ -0,0 +1,27 @@
+class ReviseEntries
```

Comment:
If I've got this right, the `ReviseEntries` exists basically to share the behavior that enables shifting between parallel and serial iterations. I suggest that the iteration behavior should be exposed in a composable approach instead so that it would be not necessary to rely on inheritance on it.

For example (a not tested snippet 😄 ):
```ruby
class ReviseFoo
  def call(entries)
    iterate_over_entries(entries) do |entry|
      # ...
    end    
  end

  private

  def iterate_over_entries(entries)
    IterateOverEntries.new.call(entries)
  end
end

class ReviseBar
  def call(entries)
    iterate_over_entries(entries) do |entry|
      # ...
    end    
  end

  private

  def iterate_over_entries(entries)
    IterateOverEntries.new.call(entries)
  end
end

class IterateOverEntries
  def call(entries, &block)
    if in_parallel?
      revise_in_parallel(entries, &block)
    else
      revise_in_sequence(entries, &block)
    end
  end

  private

  def in_parallel?
    Rails.configuration.x.parallel_requests
  end

  def revise_in_parallel(entries, &block)
    Parallel.map(entries, in_processes: 8, &block)
  end

  def revise_in_sequence(entries, &block)
    entries.map(&:block)
  end
end
```

My main point regarding to prefer composition over inheritance is that is easier to follow the revisions logic without having to look at two classes (the parent and subclass classes). Also, it can be difficult to extend some future logic by having this level of coupling usually brought by inheritance approach.